### PR TITLE
Feature/278 cleanup table component

### PR DESCRIPTION
**Pull request by brauliodiez**
Table Component was getting big and messy, I had just made a first refactoring round, got an open point I guess there's a rect that can be removed (Added a TODO Comment),

Finally table component markup gets simpler:

```tsx
    <g
      transform={`translate(${tableInfo.x}, ${tableInfo.y})`}
      onMouseDown={onMouseDown}
      className={classes.tableContainer}
    >
      <DatabaseTableBorder totalHeight={totalHeight} isSelected={isSelected} />
      <DatabaseTableHeader
        onEditTable={handleDoubleClick}
        onSelectTable={handleSelectTable}
        isSelected={isSelected}
        tableName={tableInfo.tableName}
      />
      <DatabaseTableBody renderedRows={renderedRows} />
    </g>
```

So when we need to update something we can focus on the piece that we want to update.

I have added in a separate method the rows render function (helper), will add a separate case to add unit test support to this function (this is a tought one)

I have added all you to the review just to get feedback, and to get you familiar with the codebase, ask questions as well on the PR, and if you want we can have a session to explain any doubt you may have

### CODE REVIEW

File: src/pods/canvas/components/table/database-table-border.component.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { TABLE_CONST } from '@/core/providers';
+import { TABLE_BORDER_RADIUS } from './database-table.const';
+import classes from './database-table.module.css';
```
Comment (brauliodiez) [inline src/pods/canvas/components/table/database-table-border.component.tsx:3]: We could break down CSS in several files inlcluding only the specific part for each component.

If you find this interesting we can create an issue with some guidance and assign it to a student.

File: src/pods/canvas/components/table/database-table-border.component.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+interface Props {
+  totalHeight: number;
+  isSelected: boolean;
+}
+
+export const DatabaseTableBorder: React.FC<Props> = props => {
+  const { totalHeight, isSelected } = props;
+
+  const rectStyle = {
+    filter: isSelected ? 'url(#table_component_selected)' : 'none', // Apply filter if the table is selected (highlight selected table)
```
Comment (brauliodiez) [inline src/pods/canvas/components/table/database-table-border.component.tsx:14]: Change this text to 

```
Apply SVG filter if the table is selected (highlight selected table)
```

File: src/pods/canvas/components/table/database-table-header.component.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  onEditTable: () => void;
+  isSelected: boolean;
+  tableName: string;
+  onSelectTable: () => void;
+}
+
+export const DatabaseTableHeader: React.FC<Props> = props => {
+  const { onEditTable, isSelected, tableName, onSelectTable } = props;
+
+  const pencilIconClick = (e: React.MouseEvent<SVGGElement, MouseEvent>) => {
```
Comment (brauliodiez) [inline src/pods/canvas/components/table/database-table-header.component.tsx:22]: rename this to

````
handlePencilIcon
```

File: src/pods/canvas/components/table/database-table-header.component.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        className={classes.tableHeader}
+        onClick={e => {
+          e.stopPropagation();
+        }}
+        onDoubleClick={onEditTable}
+      />
+      <TruncatedText
+        text={tableName}
+        x={TITLE_MARGIN_LEFT}
+        y={4}
```
Comment (brauliodiez) [inline src/pods/canvas/components/table/database-table-header.component.tsx:58]: Ooops, reviewing code I have just found this "Magic number", anybody knows why this 4 value?

We should know what is and encapsulate it in a meaningful constant

File: src/pods/canvas/components/table/database-table-header.component.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        <g
+          transform={`translate(${TABLE_CONST.TABLE_WIDTH - (PENCIL_ICON_WIDTH - PENCIL_MARGIN_RIGHT)}, 2)`}
+          onClick={pencilIconClick}
+        >
+          <rect
+            x="0"
+            y="0"
+            width={PENCIL_ICON_WIDTH + PENCIL_MARGIN_RIGHT}
+            fill="transparent"
+            height="25"
```
Comment (brauliodiez) [inline src/pods/canvas/components/table/database-table-header.component.tsx:73]: This is the pencil icon height, we should create a const for this (like PENCIL_ICON_WIDTH, but HEIGHT)

File: src/pods/canvas/components/table/database-table.component.tsx [hunk 6]
```diff
@@ -101,120 +97,38 @@ export const DatabaseTable: React.FC<Props> = ({
...
-          <Edit />
-        </g>
-      )}
-      ;
-      <g transform={`translate(0, ${TABLE_CONST.HEADER_TITLE_GAP})`}>
-        {renderedRows}
-      </g>
-      <rect
+      <DatabaseTableBody renderedRows={renderedRows} />
+      {/* TODO: Check with the team, is this rect neccessary? */}
```
Comment (brauliodiez) [inline src/pods/canvas/components/table/database-table.component.tsx:124]: I have remove the rect below and everything seems to be working fine, anybody know what is this rect for?

File: src/pods/canvas/components/table/database-table-header.component.tsx (hunk 4 above)
Comment (deletidev) [reply src/pods/canvas/components/table/database-table-header.component.tsx]: Yes, I know what it's for, I do it to center the text in the title background. I should have put it in a constant.

File: src/pods/canvas/components/table/components/database-table-header.component.tsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+      {/* Trick: in a div in HTML/CSS you can define border radius only for top, in SVG not, that's why
+          we draw a first rect square and a second rect with border radius */}
+      <rect
+        rx={TABLE_BORDER_RADIUS}
+        x="0"
+        y="0"
+        width={TABLE_CONST.TABLE_WIDTH}
+        height={TABLE_CONST.HEADER_HEIGHT}
+        className={classes.tableHeader}
+        onClick={e => {
```
Comment (deletidev) [inline src/pods/canvas/components/table/components/database-table-header.component.tsx:53]: Delete OnClick and onDoubleClick, we have duplicated, and this doesn't work, the important one is the rectangle at the end that has the pointer cursor as style.

**Pull request merged**